x/evm/keeper/state_transition_benchmark_test.go

```diff
@@ -0,0 +1,62 @@
+package keeper_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+func BenchmarkApplyTransaction(b *testing.B) {
+	suite := KeeperTestSuite{}
+	suite.DoSetupTest(b)
+
+	b.ResetTimer()
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+		nonce := suite.app.EvmKeeper.GetNonce(suite.address)
+		msg := evmtypes.NewTx(suite.app.EvmKeeper.ChainID(), nonce, &common.Address{}, big.NewInt(100), 21000, big.NewInt(1), nil, nil)
```

Comment:
nit, can you just set the updated nonce instead of recreating the entire msg?

Reply:
update in the new PR.
I cannot find a good way to update the nonce with the msg. Therefore, just create a template TxData structure, and update nonce, and then pass it into the eth transaction struct or native message struct.